src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -628,4 +669,50 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         // This key binding does not exist
         return nullptr;
     }
+
+    // Method Description:
+    // - Rebinds a key binding to a new key chord
+    // Arguments:
+    // - oldKeys: the key binding that we are rebinding
+    // - newKeys: the new key chord that is being used to replace oldKeys
+    // Return Value:
+    // - true, if successful. False, otherwise.
+    bool ActionMap::RebindKeys(Control::KeyChord const& oldKeys, Control::KeyChord const& newKeys)
+    {
+        const auto& cmd{ GetActionByKeyChord(oldKeys) };
+        if (!cmd)
+        {
+            // oldKeys must be bound. Otherwise, we don't know what action to bind.
+            return false;
+        }
+
+        if (newKeys)
+        {
+            // Bind newKeys
+            const auto newCmd{ make_self<Command>() };
+            newCmd->ActionAndArgs(cmd.ActionAndArgs());
+            newCmd->RegisterKey(newKeys);
+            AddAction(*newCmd);

```

Comment:
this automatically handles the case where we had the same exact command before, but with no keys, by updating it (during `AddAction`)?

Reply:
Yup! `AddAction` is smart enough to do the following:
1. check if the action existed before
   - yes: add `cmd` to `_ActionMap` for the first time (see [`_TryUpdateActionMap`](https://github.com/microsoft/terminal/blob/52560ff8182bef1eeff385cc1f329d942cd39002/src/cascadia/TerminalSettingsModel/ActionMap.cpp#L373-L377))
   - no: record `oldCmd` (aka the entry in `_ActionMap`) (and maybe record `maskingActionPair` if this was introduced in another layer). Update them as normal.